api/google_api/util/SceGoogleApiHandler.js

```diff
@@ -178,6 +178,107 @@ class SceGoogleApiHandler {
       });
     });
   }
+
+  /**
+   * Adds an event to a given calendar by its id.
+   * @param calendarId {string} calendar id for which calendar to pull from
+   * @param newEvent {Event} event to translate and add to Google Calendar
+   */
+  addEventToCalendar(calendarId, newEvent) {
+    return new Promise((resolve, reject) => {
+      const calendar =
+        google.calendar({ version: 'v3', auth: this.oAuth2Client });
+      let eventToAdd = this.translateEvent(newEvent);
+      calendar.freebusy.query({
+        resource: {
+          timeMin: eventToAdd.start.dateTime,
+          timeMax: eventToAdd.end.dateTime,
+          timeZone: eventToAdd.start.timeZone,
+          items: [{ id: 'primary' }],
+        },
+      },
+      (err, res) => {
+        if(err) resolve(err);
```

Comment:
if you `resolve` errors, the server will always send `OK`. Consider using `reject(err)` so that your code in `routes/` will make use of the `catch` block and send `BAD_REQUEST`